packages/apollo-engine-reporting/src/treeBuilder.ts

```diff
@@ -233,14 +228,19 @@ function durationHrTimeToNanos(hrtime: [number, number]) {
 
 // Convert from the linked-list ResponsePath format to a dot-joined
 // string. Includes the full path (field names and array indices).
-function responsePathAsString(p: ResponsePath | undefined) {
+function responsePathAsString(p: ResponsePath | undefined): string {
   if (p === undefined) {
     return '';
   }
-  return responsePathAsArray(p).join('.');
-}
 
-const rootResponsePath = responsePathAsString(undefined);
+  let res = String(p.key);
```

Comment:
I'd love if we left a note to future adventurers as to why we took this approach! Thoughts on prefixing with, roughly:

```suggestion
  // A previous implementation used `responsePathAsArray` from `graphql-js/execution`,
  // however, that employed an approach that created new arrays unnecessarily.
  let res = String(p.key);
```